### ISSUE TITLE
ValueError: cannot reshape array of size 

### ISSUE THREAD

**Issue by rtmatx** [pyviz/datashader#774]
```
import pandas.util.testing as tm
tm.N, tm.K = 150000, 30  # Module-level default rows/columns
df = tm.makeTimeDataFrame(freq='1Min')
df.index.name = 'timestamp'
df.hvplot.scatter(x='A', y='B', datashade=True, dynspread=True)
```

Hi,
 
The above code produces the error `ValueError: cannot reshape array of size ` error at 

```
   1123         shape = img.shape
   1124         flat_shape = np.multiply.reduce(shape[:2])
-> 1125         rgb = img.reshape((flat_shape, 4)).view('uint32').reshape(shape[:2])
   1126         return rgb
   1127 
```

I have latest numpy installed (1.16.4) on my system. Did anyone face this issue before ?

Thanks
Ravi

**Comment by jbednar** [comment]
I  can't quite follow what is going on, but I can reproduce it:

```
import pandas.util.testing as tm, hvplot.pandas
tm.N, tm.K = 1500, 30  # Module-level default rows/columns
df = tm.makeTimeDataFrame(freq='1Min')
df.index.name = 'timestamp'
df.hvplot.scatter(x='A', y='B', datashade=True, dynspread=False)
```

![image](https://user-images.githubusercontent.com/1695496/62184306-10e23c00-b323-11e9-9f21-fe528b36cfae.png)

It doesn't look like a datashader issue to me, but one in HoloViews or hvPlot.  Oddly, the plot renders but looks crazy when I turn off dynspread:

```
df.hvplot.scatter(x='A', y='B', datashade=True, dynspread=False)
```
![image](https://user-images.githubusercontent.com/1695496/62184361-5dc61280-b323-11e9-9ecf-c9d11d0acf6a.png)

which behaves very strangely on zoom:

![image](https://user-images.githubusercontent.com/1695496/62184390-7a624a80-b323-11e9-804b-291aab201ce1.png)
![image](https://user-images.githubusercontent.com/1695496/62184401-85b57600-b323-11e9-9adb-ab7468be15f0.png)

Yet it looks reasonable when datashade=False:

```
df.hvplot.scatter(x='A', y='B', datashade=False, dynspread=False)
```
![image](https://user-images.githubusercontent.com/1695496/62184422-a2ea4480-b323-11e9-958e-85dffe7a732c.png)

@philippjfr , any ideas?

**Comment by rtmatx** [comment]
I am not sure if can help, but when I try to adjust the limits for the canvas, I get few warnings

![image](https://user-images.githubusercontent.com/23462596/62488397-d563c980-b789-11e9-9602-fc6088455f58.png)

**Comment by philippjfr** [comment]
Yes, this is caused by `kdims` that are called one of the following 'R', 'G', 'B' or 'A'. HoloViews ends up confusing the value dimensions with the same names with those key dimensions and either breaks or renders some bizarre colors.

**Comment by rtmatx** [comment]
Thanks. I can confirm that it is working. So the column names shouldn't be `R`, `G`, `B`, `A`. 

![image](https://user-images.githubusercontent.com/23462596/62510098-6c08a880-b7d3-11e9-98fe-f14de7c86f46.png)

**Comment by jbednar** [comment]
Tricky!  Thanks for tracking that down,  @philippjfr.  What should we do about it?

1. Detect when it is happening and raise an exception?  That's an improvement for notebook users, but it seems problematic for no-code interactive tools like `dfviz` that are meant to be able to work with arbitrary dataframes that might have those column names.
2. Print a loud warning saying that hv.RGB can't be used with dataframes with those columns, but display the plot anyway?  Would be confusing for users and doesn't really solve the problem.
3. Rename the original columns that conflict, inside the hv.shade operation?  Seems like this will cause problems with hover, linking, selection, etc., so it would still presumably need warnings.
4. Rename the columns of hv.RGB objects to something that would rarely occur, e.g. RGBA_R, RGBA_G, etc.?  Seems like that would probably require backwards-incompatible changes in Bokeh.

All of those options seem problematic, so hopefully I'm missing some better option #5.

**Comment by ppwadhwa** [comment]
@philippjfr @jlstevens any opinions?

**Comment by jlstevens** [comment]
I agree this is tricky. 

My only thought would be that an xarray could label the RGB(A) channels correctly without needing us to generate those names ourselves. That said, losing the ability to generate RGB(A) elements from say numpy arrays would be quite annoying!

My only suggestion right now is to document this and make people aware of the potential problems with a name clash. Looking at the `RGB` element you should be able to specify your own dimension objects with different names for instance...
